### ISSUE TITLE
Huge performance concern for directives

### ISSUE THREAD

**Issue by ProLoser** [luisfarzati/angulartics#401]
So I was poking around the code for unrelated reasons and came across this line: https://github.com/luisfarzati/angulartics/blob/master/src/angulartics.js#L260

You are essentially adding watchers for every single property of an element that has analytics. This is an insane amount of watchers and is completely unnecessary until the event tracking trigger is actually initiated. You need to change this to be evaluations that perform at event-track time and not as constant watchers that update data that may very well never get used (if the tracking event never even fires).

**Comment by luisfarzati** [comment]
Yes, that's a very good point. It seems this was introduced here: https://github.com/luisfarzati/angulartics/commit/6a925390758c5d6e5e013dfa60d5a76e6446f58f. @pcwinters do you remember if there was any specific reason to watch this all the time? Otherwise I guess it's safe to assume evaluation only at event-track time, as @ProLoser suggests.

**Comment by pcwinters** [comment]
Sorry for the delay, @luisfarzati. I'd like to believe it was intentional, but it's been over a year. The commit sure seems intentional. At the time I was working on a rather large and deep angular app, and I believe there was an issue with analytics properties as expressions.

I'd need an angular refresher, but maybe this was the only issue?
```properties[name.slice(9).toLowerCase()] = $attrs[name];```
